### PR TITLE
Dependency upgrade & support for spacy 2.0

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- test upper / lower case behaviour of spacy model
- reformatted evauluation tests

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,34 +4,20 @@ cache:
...
 - pip install git+https://github.com/tmbo/MITIE.git
 - pip install -r alt_requirements/requirements_dev.txt
 - pip install -e .
-- pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-1.2.0/en_core_web_sm-1.2.0.tar.gz
+- pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_md-2.0.0/en_core_web_md-2.0.0.tar.gz
   --no-cache-dir > jnk
-- python -m spacy link en_core_web_sm en
-- pip install https://github.com/explosion/spacy-models/releases/download/de_core_news_md-1.0.0/de_core_news_md-1.0.0.tar.gz
+- python -m spacy link en_core_web_md en
+- pip install https://github.com/explosion/spacy-models/releases/download/de_core_news_sm-2.0.0/de_core_news_sm-2.0.0.tar.gz
```
Comment (akelad) [inline .travis.yml]: so have we checked the German model works well? given there's only a small one for that for some reason...

Comment (tmbo) [reply .travis.yml]: I didn't check the models intent classification performance for german / french.

Comment (akelad) [reply .travis.yml]: well idk - do we want to mention that in the tutorial somewhere?

Comment (tmbo) [reply .travis.yml]: Just checked it on two datasets and there wasn't any difference - which doesn't say there is none but I think we are good to go for now :+1:

**Pull request merged**